lib/elixir/lib/file.ex

```diff
@@ -544,6 +580,27 @@ defmodule File do
     write_stat!(dest, %{stat!(dest) | mode: stat!(src).mode})
   end
 
+  # Most of the underlying move functionality delgated
+  # to elrang's rename function
+  # http://www.erlang.org/doc/man/file.html#rename-2
+  defp do_mv_file(src, dest, callback) do
+    if File.exists?(dest) && !callback.(src,dest) do
```

Comment:
Another option is to simply drop the callback aspect and ensure we move it even if the target exists, which is what both Erlang and `mv` do! What do you think?